dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/ror/RorIndexingService.java

```diff
@@ -0,0 +1,49 @@
+package edu.harvard.iq.dataverse.search.ror;
+
+import edu.harvard.iq.dataverse.persistence.ror.RorData;
+import edu.harvard.iq.dataverse.search.RorSolrClient;
+import io.vavr.control.Try;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.response.UpdateResponse;
+import org.apache.solr.common.SolrInputDocument;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Service dedicate for indexing ROR data.
+ */
+@Stateless
+public class RorIndexingService {
+
+    private static final Logger logger = Logger.getLogger(RorIndexingService.class.getCanonicalName());
+
+    @Inject
+    @RorSolrClient
+    private SolrClient solrServer;
+
+    public UpdateResponse indexRorRecord(RorData rorData) {
+        assert rorData.getId() != null;
+
+        SolrInputDocument solrInputDocument = new SolrInputDocument();
+
+        solrInputDocument.addField(RorSolrField.ID.getExactName(), rorData.getId());
```

Comment:
IMO it would be safer to assume that ror is unique identifier of collection instead of database id. The way we are importing ror's could introduce some duplicates in some more uncommon scenarios